Automattic/wp-calypso
# Setting write your first post as done in the Launchpad and avoiding redirect out of it

This is a follow up of https://github.com/Automattic/wp-calypso/pull/75734 and https://github.com/Automattic/wp-calypso/pull/75649 

Video of the proposed changes in action:

https://user-images.githubusercontent.com/1044309/232881433-86ecda59-34f6-4730-aab8-847ff3fabaee.mp4



## Proposed Changes

* Avoiding redirect out from the Launchpad
* Marks `Write your first post` as completed in the launchpad 

## Testing Instructions




* Apply this branch on your local Calypso
* Make sure `widgets.wp.com` is sandboxed.
* Follow the [wpcom-block-editor sync flow](https://github.com/Automattic/wp-calypso/tree/trunk/apps/wpcom-block-editor#dev-workflow). In summary: 
  - cd apps/wpcom-block-editor/src 
  - yarn dev --sync

* Create a new user using `/setup/start-writing/`
* After publishing a post, you should get redirected to the Launchpad and stay there
* Make sure the `Write your first post` option in the lateral menu of the Launchpad is marked as done
* Make sure tests passes with this command: `yarn run jest apps/wpcom-block-editor/src/wpcom/features/test/redirect-onboarding-user-after-publishing-post.test.js`

## Pre-merge Checklist
